src/components/plots/barPlot.ts

```diff
@@ -43,6 +43,67 @@ export module Plot {
       this._baseline = this._renderArea.append("line").classed("baseline", true);
     }
 
+    protected _onDatasetUpdate() {
+      super._onDatasetUpdate();
+      if (this._isAnchored) {
+        this._updateBarExtent();
+      }
+    }
+
+    public detach() {
+      super.detach();
+      this._datasetKeysInOrder.forEach((key) => {
+        var scaleKey = this.getID().toString() + "_" + key;
+        this._xScale._removeExtent(scaleKey, "bar-extent");
+      });
+      return this;
+    }
+
+    public remove() {
+      super.remove();
+      this._datasetKeysInOrder.forEach((key) => {
+        var scaleKey = this.getID().toString() + "_" + key;
+        this._xScale._removeExtent(scaleKey, "bar-extent");
+      });
+      return this;
```

Comment:
`remove()` calls `detach()` (see `AbstractComponent`), so you don't need to override this.